torch_xla/csrc/random.cpp

```diff
@@ -69,6 +69,29 @@ xla::Shape MakeRngShape(const xla::Shape& shape) {
 
 }  // namespace
 
+xla::XlaOp RngDiscreteUniform(xla::XlaOp seed, const xla::Shape& shape,
+                              xla::XlaOp minval, xla::XlaOp maxval) {
+  xla::PrimitiveType minval_type = XlaHelpers::TypeOfXlaOp(minval);
+  xla::PrimitiveType maxval_type = XlaHelpers::TypeOfXlaOp(maxval);
+  xla::XlaOp rng_seed = MakeSeed(seed);
+  xla::XlaOp initial_state =
+      xla::Zero(rng_seed.builder(), xla::PrimitiveType::U64);
+  xla::Shape rng_shape(shape);
+  XLA_CHECK_EQ(minval_type, maxval_type);
```

Comment:
I'd move these checks soon after you got `maxval_type` above.

Reply:
@JackCaoG I think, since these checks only use min/max types, better move them up at line 76.